src/main/java/cubicchunks/server/ServerCubeCache.java

```diff
@@ -279,9 +274,11 @@ public Cube getCube(int cubeX, int cubeY, int cubeZ, @Nonnull Requirement req) {
 	 */
 	private void onCubeLoaded(@Nullable Cube cube, @Nonnull Column column) {
 		if(cube != null) {
-			column.addCube(cube);
 			cubeMap.put(cube); // cache the Cube
-			cube.onLoad();             // init the Cube
+			if(!column.getLoadedCubes().contains(cube)) {
+				column.addCube(cube);
+				cube.onLoad(); // init the Cube
+			}
```

Comment:
I don't think it's possible, cube can be saved only once it's loaded. And only saving a cube can modify one that is being loaded. This is only to prevents multiple callbacks for the same queued load from adding the same cube twice. Probably should add comment explaining that.